### PR TITLE
feat(cart): add place order reducers

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[ ] No
```

## Other information

### CODE REVIEW

File: libs/cart/src/reducers/cart-order-state.interface.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export interface DaffCartOrderReducerState {
+  order: {id: string | number};
```
Comment (lderrickable) [inline libs/cart/src/reducers/cart-order-state.interface.ts:2]: seems weird to make this an object, but not make it generic. I think we should go with one or the other. Probably make it generic to future proof it.

Comment (griest024) [reply libs/cart/src/reducers/cart-order-state.interface.ts:2]: Making it generic would introduce a dependency onto checkout. Maybe thats a reason to move the place order driver into checkout instead.

Comment (griest024) [reply libs/cart/src/reducers/cart-order-state.interface.ts]: changed

File: libs/cart/src/reducers/cart-order-state.interface.ts [hunk 2]
```diff
@@ -0,0 +1,7 @@
+import { DaffCartOrderResult } from '../models/cart-order-result';
+
+export interface DaffCartOrderReducerState<T extends DaffCartOrderResult = DaffCartOrderResult> {
+  order: T;
```
Comment (lderrickable) [inline libs/cart/src/reducers/cart-order-state.interface.ts:4]: I think this should be called something else to be more clear that it isn't actually an order. Maybe just cartOrderResult?

Comment (griest024) [reply libs/cart/src/reducers/cart-order-state.interface.ts]: changed

File: libs/cart/src/reducers/cart-reducers-state.interface.ts [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import { DaffCartReducerState } from './cart-state.interface';
 import { DaffCart } from '../models/cart';
+import { DaffCartOrderReducerState } from './cart-order-state.interface';
 
 export interface DaffCartReducersState<T extends DaffCart = DaffCart> {
-  cart: DaffCartReducerState<T>
+  cart: DaffCartReducerState<T>,
+  order: DaffCartOrderReducerState
```
Comment (lderrickable) [inline libs/cart/src/reducers/cart-reducers-state.interface.ts:7]: This should be generic

Comment (griest024) [reply libs/cart/src/reducers/cart-reducers-state.interface.ts]: done

**Pull request merged**